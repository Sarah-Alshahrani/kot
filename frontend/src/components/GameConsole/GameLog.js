import React from "react";
import "./GameConsole";

class GameLog extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      username: props.currentUser,
      gameRoom: props.currentRoom,
      log: props.data
    };
  }

  componentDidMount() {
    this.scrollToBottom();
  }

  componentDidUpdate() {
    this.scrollToBottom();
  }

  scrollToBottom() {
    const chat = this.messagesEnd;

    if (chat) {
      const scrollHeight = chat.scrollHeight;
      const height = chat.clientHeight;
      const maxScrollTop = scrollHeight - height;
      chat.scrollTop = maxScrollTop > 0 ? maxScrollTop : 0;
    }
  }

  renderMessages(messages) {
    if (messages && messages.length > 0) {
      const currentUser = this.state.username;
      return messages.map((message, i) => (
        <li key={i} className={message.user === currentUser ? "me" : "him"}>
          <p>{message}</p>
        </li>
      ));
    }
  }

  render() {
    const game_messages = this.props.data;
    return (
      <div className="container">
        <ul
          ref={el => {
            this.messagesEnd = el;
          }}
        >
          {game_messages && this.renderMessages(game_messages)}
        </ul>
      </div>
    );
  }
}

export default GameLog;
