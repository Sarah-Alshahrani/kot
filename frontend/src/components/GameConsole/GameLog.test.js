import React from 'react';
import { shallow } from 'enzyme';

import GameLog from './GameLog';

describe('Verify GameLog layout template', () => {
    it('Verify parameter provided logs are rendered correctly', () => {

        const tLog = [];

        const component = shallow(<GameLog data={tLog} />);
        expect(component).toMatchSnapshot();
    });
})