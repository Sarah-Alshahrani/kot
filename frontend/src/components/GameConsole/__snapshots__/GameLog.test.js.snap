// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify GameLog layout template Verify parameter provided logs are rendered correctly 1`] = `
<div
  className="container"
>
  <ul />
</div>
`;
